src/compile.ts

```diff
@@ -220,14 +220,45 @@ export function compile(
       }
 
       function visitAppsyncResolver(call: ts.NewExpression): ts.Node {
-        if (call.arguments?.length === 1) {
-          const impl = call.arguments[0];
-          if (ts.isFunctionExpression(impl) || ts.isArrowFunction(impl)) {
+        if (call.arguments?.length === 3) {
+          const [scope, id, props] = call.arguments;
+          if (ts.isObjectLiteralExpression(props)) {
```

Comment:
Need a friendly error when not an object?

Reply:
Decided against this as it breaks our tests. Will instead rely on a static validation error. Also, moved the function to the 4th argument.

Reply:
It breaks our tests because we have a util that passes the FunctionDecl and casts to any. This breaks.